test helper in TS template

Add the `require('fastify-cli/helper')` utility to the TS template.
Now it exports the old interface.

The test was failing during the implementation:

```
templates/app-ts/test/routes/example.test.ts 2> TSError: ⨯ Unable to compile TypeScript:
templates/app-ts/test/routes/example.test.ts 2> templates/app-ts/test/helper.ts(3,23): error TS7016: Could not find a declaration file
for module 'fastify-cli/helper'. '/Users/mspigolon/workspace/fastify-cli/node_modules/fastify-cli/helper.js' implicitly has an 'any'
type.
```

_Originally posted by @Eomm in https://github.com/fastify/fastify-cli/issues/446#issuecomment-1012562061_

> Add the `require('fastify-cli/helper')` utility to the TS template. Now it exports the old interface.
> 
> The test was failing during the implementation:
> 
> ```
> templates/app-ts/test/routes/example.test.ts 2> TSError: ⨯ Unable to compile TypeScript:
> templates/app-ts/test/routes/example.test.ts 2> templates/app-ts/test/helper.ts(3,23): error TS7016: Could not find a declaration file
> for module 'fastify-cli/helper'. '/Users/mspigolon/workspace/fastify-cli/node_modules/fastify-cli/helper.js' implicitly has an 'any'
> type.
> ```
> 
> _Originally posted by @Eomm in [#446 (comment)](https://github.com/fastify/fastify-cli/pull/446#issuecomment-1012562061)_

I was trying to reproduce the failing test, but it seems that any change made to templates/app-ts/test/routes/example.test.ts does not trigger any error.

--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
   "scripts": {
     "lint": "standard",
     "lint:fix": "standard --fix",
-    "unit": "tap \"test/**/*.test.{js,ts}\" \"templates/**/*.test.{js,ts}\" --timeout 120",
+    "unit": "tap \"test/**/*.test.{js,ts}\" \"templates/**/*.test.{js,ts}\" --timeout 300",
     "pretest": "xcopy /e /k /i . \"..\\node_modules\\fastify-cli\" || rsync -r --exclude=node_modules ./ node_modules/fastify-cli || echo 'this is fine'",
     "test": "npm run lint && npm run unit && npm run test:typescript",
     "test:typescript": "tsd templates/plugin && tsc --project templates/app-ts/tsconfig.json && del-cli templates/app-ts/dist"

--- a/templates/app-ts/test/helper.ts
+++ b/templates/app-ts/test/helper.ts
@@ -1,10 +1,11 @@
 // This file contains code that we reuse between our tests.
-import Fastify from 'fastify'
-import fp from 'fastify-plugin'
-import App from '../src/app'
+const helper = require('fastify-cli/helper.js')
+import * as path from 'path'
 import * as tap from 'tap';
 
 export type Test = typeof tap['Test']['prototype'];
+
+const AppPath = path.join(__dirname, '..', 'src', 'app.ts')
 
 // Fill in this config with all the configurations
 // needed for testing the application
@@ -14,14 +15,13 @@ async function config () {
 
 // Automatically build and tear down our instance
 async function build (t: Test) {
-  const app = Fastify()
+  // you can set all the options supported by the fastify CLI command
+  const argv = [AppPath]
 
   // fastify-plugin ensures that all decorators
   // are exposed for testing purposes, this is
   // different from the production setup
-  void app.register(fp(App), await config())
-
-  await app.ready();
+  const app = await helper.build(argv, await config())
 
   // Tear down our app after we are done
   t.teardown(() => void app.close())

